bin/watch-temporal-worker

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+"""
+File watcher script for temporal workers that restarts the worker when Python files change.
+"""
+
+import logging
+import sys
+import time
+import signal
+import subprocess
+import threading
+from pathlib import Path
+from typing import Optional
+
+
+# Configure logging to output to console
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    handlers=[logging.StreamHandler(sys.stdout)]
+)
+logger = logging.getLogger(__name__)
+
+
+class FileWatcher:
+    def __init__(self, paths_to_watch: list[str], extensions: set[str] | None = None):
+        self.paths_to_watch = [Path(p).resolve() for p in paths_to_watch]
+        self.extensions = extensions or {'.py'}
```

Comment:
We seem to be watching every `.py` file for all workers, but workers generally care only about very specific files. For example, for batch export workers, we can safely ignore any changes outside `products/batch_exports`, `posthog/batch_exports` and `posthog/temporal`.

Watching every `.py` file for every worker looks very wasteful. Maybe I am reading this wrong?